crates/rune/src/compile/v1/assemble.rs

```diff
@@ -1957,9 +1957,38 @@ fn expr_object<'hir>(
 ) -> compile::Result<Asm<'hir>> {
     let guard = cx.scopes.child(span)?;
 
-    for assign in hir.assignments {
+    let mut order = Vec::with_capacity(hir.assignments.len());
+
+    for assign in hir.assignments.iter() {
         expr(cx, &assign.assign, Needs::Value)?.apply(cx)?;
         cx.scopes.alloc(&span)?;
+
+        let position = assign.position.ok_or(compile::Error::msg(
+            span,
+            format!("missing position for field assignment {}", assign.key.1),
+        ))?;
```

Comment:
Since compile::Error::msg isn't a constant op the preferred way to do this would be to use `ok_or_else`, or the now nicer let-else:
```rust
let Some(position) = assign.position else {
    return Err(compile::Error::msg(
        span,
        format_args!("Missing position for field assignment {}", assign.key.1),
    ));
};
```